lib/proxy/peer/client.go

```diff
@@ -596,11 +602,12 @@ func (c *Client) connect(id string, proxyPeerAddr string) (*clientConn, error) {
 	connCtx, cancel := context.WithCancel(c.ctx)
 	wg := new(sync.WaitGroup)
 
-	transportCreds := newProxyCredentials(credentials.NewTLS(tlsConfig))
+	expectedPeer := peerID + "." + c.config.ClusterName
```

Comment:
Could we add a comment referencing where this was coming from?
https://github.com/gravitational/teleport/blob/3b07af228a77324f9eff99efc8eac43165dca5c4/api/utils/cluster.go#L29-L33

I'm unfamiliar with this area of the code so I had to hunt to find it.

I assume the peer ID will be decoded from hex at some point before its compared with this as well?

Reply:
actually I'm not even sure I found the right place? This code seems to encode "<cluster>.<domain>", but maybe it's just a difference of terminology?

Reply:
thanks for clarifying